pkg/virtctl/vm/add_volume.go

```diff
@@ -163,13 +167,26 @@ func addVolume(vmiName, volumeName, namespace string, virtClient kubecli.Kubevir
 			return fmt.Errorf("error adding volume, invalid cache value %s", cache)
 		}
 	}
-	if !persist {
-		err = virtClient.VirtualMachineInstance(namespace).AddVolume(context.Background(), vmiName, hotplugRequest)
-	} else {
-		err = virtClient.VirtualMachine(namespace).AddVolume(context.Background(), vmiName, hotplugRequest)
+	retry := 0
+	for retry < maxRetries {
+		if !persist {
+			err = virtClient.VirtualMachineInstance(namespace).AddVolume(context.Background(), vmiName, hotplugRequest)
+		} else {
+			err = virtClient.VirtualMachine(namespace).AddVolume(context.Background(), vmiName, hotplugRequest)
+		}
+		if err != nil && err.Error() != concurrentError {
```

Comment:
I can't it is not returning a 422. So IsInvalid will not return the desired result.

Reply:
Ah sorry I thought this was being reported back from the client via `NewGenericServerResponse` below:

https://github.com/kubevirt/kubevirt/blob/f85a8117b8a90fd913ec0719faae9506866d1525/vendor/k8s.io/apimachinery/pkg/api/errors/errors.go#L474-L476

Reply:
Unfortunately no. As I said above I don't want to make too many changes to this at this point since we will be replacing the entire mechanism to align with other hotplug options (cpu/mem/network) to be declarative instead of what we have now. This is why I choice to go with the simple retry in the client option instead of some mechanism on the server side.